libr/anal/xrefs.c

```diff
@@ -215,30 +215,33 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
 	RList *list = r_anal_ref_list_new();
 	listxrefs (anal->dict_refs, UT64_MAX, list);
 	if (rad == 'j') {
-		anal->cb_printf ("{");
+		anal->cb_printf ("[");
 	}
 	r_list_foreach (list, iter, ref) {
 		int t = ref->type ? ref->type: ' ';
 		switch (rad) {
 		case '*':
-			anal->cb_printf ("ax%c 0x%"PFMT64x" 0x%"PFMT64x"\n",
-				t, ref->addr, ref->at);
+			anal->cb_printf ("ax%c 0x%"PFMT64x" 0x%"PFMT64x"\n", t, ref->addr, ref->at);
 			break;
 		case '\0':
 			{
 				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
-				r_str_replace_char (name, ' ', 0);
-				anal->cb_printf ("%40s", name? name: "");
-				free (name);
+				if (name) {
+					r_str_replace_ch (name, ' ', 0, true);
+					anal->cb_printf ("%40s", name);
+					free (name);
+				} else {
+					anal->cb_printf ("%40s", "(null)");
```

Comment:
wtf why are you printing null with 40 spaces of padding?? i would prefer to not show anything in that case, (null). i mean if you want to show null just use r_str_get.. but i think this is getting overengineered

Reply:
because i didn't wanted to change the behavior of the code. before it was displaying empty line with 40 spaced chars.